src/NFTAuction.sol

```diff
@@ -101,7 +93,11 @@ contract NFTAuctionMachine is
     @dev Allows users to bid & send eth to the contract.
     */
     function bid() public payable nonReentrant {
-        if (block.timestamp > auctionEndingAt) {
+        if (nft.isMaxSupplyReached()) {
+            revert MAX_SUPPLY_REACHED();
```

Comment:
maybe more gas efficient to only check this if its the first bid in a new auction

Reply:
yeah